simulator/source/Combat_simulator.cpp

```diff
@@ -36,13 +36,12 @@ Combat_simulator::Combat_simulator(const Combat_simulator_config& config) : conf
 
 void Combat_simulator::add_talent_effects(const Character& character)
 {
-    execute_rage_cost_ = std::vector<int>{15, 13, 10}[character.talents.improved_execute];
-    heroic_strike_rage_cost_ = 15 - character.talents.improved_heroic_strike;
-    whirlwind_rage_cost_ = 25;
-    mortal_strike_rage_cost_ = 30;
-    bloodthirst_rage_cost_ = 30;
-
-    tactical_mastery_rage_ = 10 + character.talents.tactical_mastery * 5;
+    heroic_strike_rage_cost_ = 15 - character.talents.improved_heroic_strike - character.talents.focused_rage;
+    execute_rage_cost_ = std::vector<int>{15, 13, 10}[character.talents.improved_execute] - character.talents.focused_rage;
+    whirlwind_rage_cost_ = 25 - character.talents.focused_rage;
+    mortal_strike_rage_cost_ = 30 - character.talents.focused_rage;
+    bloodthirst_rage_cost_ = 30 - character.talents.focused_rage;
+    devastate_rage_cost_ = 15 - character.talents.focused_rage - character.talents.improved_sunder_armor;
```

Comment:
there's a sunder_armor() already, which should also use these talents (for good measure) - or it could just share the devastate_rage_cost_

Reply:
Added a variable both for sunder armor cost and hamstring cost and updated the actual skill's logic to include them.